io-engine/src/bin/io-engine-client/v1/replica_cli.rs

```diff
@@ -300,8 +300,15 @@ async fn replica_list(
                     let size = ctx.units(Byte::from_bytes(r.size.into()));
                     let capacity = ctx
                         .units(Byte::from_bytes(usage.capacity_bytes.into()));
-                    let allocated = ctx
-                        .units(Byte::from_bytes(usage.allocated_bytes.into()));
+                    let allocated = if r.is_snapshot {
```

Comment:
This means that we display size of all predecessor snapshots and don't take into account how much space the snapshot itself occupies ?

Reply:
> This means that we display size of all predecessor snapshots and don't take into account how much space the snapshot itself occupies ?

This logic to be reverted back. By Clearing blob->num_used_clusters_cache for thin provisioned replica, the snapshot and replica allocated size is coming proper.